### ISSUE TITLE
Fill Gaps Between Walls: wall and skin not actually filling tiny features.

### ISSUE THREAD

**Issue by JohnEdwa** [smartavionics/Cura#40]
Slicing this tiny cable clip where there is no space for a full wall, only "Everywhere" actually fills the gap successfully even though according to the preview the fill is done as a shell top/bottom layer, and as I'm asking for 3 walls it should be a thin inner wall, not skin (so it should be filled even with "Only Wall Gaps" setting).

Project file: [CE3_Cable_Clip.zip](https://github.com/smartavionics/Cura/files/4129399/CE3_Cable_Clip.zip)

![wallandskin](https://user-images.githubusercontent.com/5417183/73379659-57612500-42cb-11ea-9a85-589269542233.PNG)
![everywhere](https://user-images.githubusercontent.com/5417183/73379663-592ae880-42cb-11ea-8ce0-87415e591c2c.PNG)

**Application Version**
20200125

Also, a side note, the explanation help popup for this quite confusing. My suggestion would be to completely re-order and reword it to something such as:

"Fills the gaps between walls where no walls fit. 
Only Wall Gaps fills the gaps between adjacent wall lines.
Wall and Skin Gaps fills the gap between walls, and between a wall and the skin.
Everywhere fills the gaps between walls, a wall and the skin, and a wall and the infill."

(Although I'm a non-native English speaker/writer so I'm probably not the best for this)

**Comment by smartavionics** [comment]
Yes, it's not working right. I will investigate, thanks for the report.

**Comment by smartavionics** [comment]
I have removed the wall and skin gaps option as it wasn't working right and I can't be bothered to fix it. So the next release will have options: nowhere, only wall gaps and everywhere.

**Comment by JohnEdwa** [comment]
Not ideal, but understandable. I'll continue using everywhere, as the main reason I want it enabled is for filling visible gaps on the top layers which this impementation of "walls only" will leave unfilled.

**Comment by 5axes** [comment]
Thank's for your contribution @smartavionics , this is with this type of example that I know why I 'using your release rather than the official.
![4 4 versus Master](https://user-images.githubusercontent.com/11015345/73461430-6b179480-437a-11ea-9d87-435fa6e000f6.jpg)

**Comment by smartavionics** [comment]
You're welcome.

What I really don't understand is why UM do not fix some of the long standing issues that Cura suffers from. Sure, they have to put most of their effort into stuff that is relevant to UM printers as that's how they get paid but a lot of the worst bugs/features in Cura affect all printer types. So this particular issue (poor quality gap filling and thin line printing) along with Cura's poor handling of walls whose width is not close to an even number of line width should have been fixed years ago but they don't seem to care.

**Comment by noisettetbou** [comment]
Hi. I use curamaster 20200125 and the option "fill gaps between walls" works. I think you've got a little bug but the idea is well and I hope this option is back in a future release.

**Comment by smartavionics** [comment]
I just removed the Wall And Skin Gaps option, the other options are still there.
